### PR TITLE
chapter13_user_microposts

### PR DESCRIPTION
**Pull request by tenyiting**
chapter13_user_microposts

### CODE REVIEW

**Comment by tenyiting** [comment]
ready

**Comment by tenyiting** [comment]
ready

**Comment by tenyiting** [comment]
ready

File: app/assets/javascripts/microposts.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (ghost) [inline app/assets/javascripts/microposts.coffee]: Delete this file please

File: app/assets/stylesheets/microposts.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Microposts controller here.
```
Comment (ghost) [inline app/assets/stylesheets/microposts.scss]: Delete this file please

File: app/helpers/microposts_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module MicropostsHelper
```
Comment (ghost) [inline app/helpers/microposts_helper.rb]: Delete this file please

File: test/controllers/microposts_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (ghost) [inline test/controllers/microposts_controller_test.rb]: Delete this file please

File: test/fixtures/microposts.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+one:
```
Comment (ghost) [inline test/fixtures/microposts.yml]: Delete this file please

File: test/models/micropost_test.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (ghost) [inline test/models/micropost_test.rb]: Delete this file please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by tenyiting** [comment]
ready

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -3,33 +3,41 @@ class UsersController < ApplicationController
...
 
 
- def index
+  def index
     @users = User.paginate(page: params[:page])
   end
 
-  def show; end
+  def show
+    @user = User.find_by id: params[:id]
```
Comment (huongnt-2545) [inline app/controllers/users_controller.rb:12]: You should check the exist of user before getting user's microposts

```suggestion
    @user = User.find_by id: params[:id]
    if @user.nil?
      flash[:danger] = "Not found user"
      redirect_to root_url
    end
```

File: app/models/micropost.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+class Micropost < ApplicationRecord
+  belongs_to :user
+  default_scope -> { order(created_at: :desc) }
+  has_one_attached :image
+  validates :user_id, presence: true
+  validates :content, presence: true, length: {maximum: 150}
+  validates :image, content_type: { in: %w[image/jpeg image/gif image/png],
```
Comment (huongnt-2545) [inline app/models/micropost.rb]:
```suggestion
  validates :image, content_type: {in: %w[image/jpeg image/gif image/png]
    , message: "must be a valid image format" }
    , size: {less_than: 5.megabytes, message: "should be less than 5MB"}
```

File: app/controllers/users_controller.rb (hunk 7 above)
Comment (tenyiting) [reply app/controllers/users_controller.rb:12]: Hi.. what do you mean? is it delete that line?

Comment (tenyiting) [reply app/controllers/users_controller.rb:12]: I just updated. pls check again. Thank you!

**Comment by tenyiting** [comment]
ready

File: app/models/micropost.rb [hunk 9]
```diff
@@ -0,0 +1,73 @@
+class User < ApplicationRecord
```
Comment (huongnt-2545) [inline app/models/micropost.rb]: why you have `User` class in this file???

**Comment by tenyiting** [comment]
ready

File: app/controllers/users_controller.rb [hunk 10]
```diff
@@ -4,12 +4,19 @@ class UsersController < ApplicationController
...
 
-
- def index
+  def index
     @users = User.paginate(page: params[:page])
   end
 
-  def show; end
+  def show
+ @user = User.find_by id: params[:id]
```
Comment (huongnt-2545) [inline app/controllers/users_controller.rb]: indent
```suggestion
    @user = User.find_by id: params[:id]
```

**Review by huongnt-2545 — APPROVED** [review]
merged
I have a small comment related the indent, you can fix in next pull
You can merge this pull

File: app/controllers/users_controller.rb (hunk 10 above)
Comment (tenyiting) [reply app/controllers/users_controller.rb]: i just fixed it. Thank you!!

**Pull request merged**